src/chunk_constraint.c

```diff
@@ -253,12 +260,17 @@ ts_chunk_constraints_add_from_tuple(ChunkConstraints *ccs, const TupleInfo *ti)
 		hypertable_constraint_name = DatumGetName(DirectFunctionCall1(namein, CStringGetDatum("")));
 	}
 
+	/* check if it's a correlated constraint */
+	if (strncmp(CC_DIM_PREFIX, NameStr(*constraint_name), strlen(CC_DIM_PREFIX)) == 0)
+		correlated = true;
```

Comment:
Since you're doing this check in more places would be good to encapsulate it in an static inline function.

Reply:
Given that there are now at least 3 types of constraints in our metadata, maybe we should have an enum/type for this?

So, instead of `is_dimension_constraint` we can have a function `dimension_type()` and you can `switch` on that. An enum also makes it explicit what type of dimensions there is instead of it being somewhat implicit.